core/src/main/java/org/mapstruct/AnnotateWith.java

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright MapStruct Authors.
+ *
+ * Licensed under the Apache License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+package org.mapstruct;
+
+import java.lang.annotation.Annotation;
+import java.lang.annotation.Repeatable;
+import java.lang.annotation.Retention;
+import java.lang.annotation.Target;
+
+import static java.lang.annotation.ElementType.ANNOTATION_TYPE;
+import static java.lang.annotation.ElementType.METHOD;
+import static java.lang.annotation.ElementType.TYPE;
+import static java.lang.annotation.RetentionPolicy.SOURCE;
+
+/**
+ * This can be used to have mapstruct generate additional annotations on classes/methods.
+ *
+ * @author Ben Zegveld
+ * @since 1.5
+ */
+@Repeatable( AnnotateWiths.class )
+@Retention( SOURCE )
+@Target( { TYPE, METHOD, ANNOTATION_TYPE } )
+public @interface AnnotateWith {
+    Class<? extends Annotation> value();
+
+    Parameter[] parameters() default {};
+
+    @interface Parameter {
+        String key();
```

Comment:
You can have Annotations within Annotations.
For example: `@Mapper( builder = @Builder)`
This will not be supported initially, because I simply can't yet predict how to correctly analyze and generate this. Having the first iteration without support for these kind of annotations feels like a good start.

Reply:
Got it. We don't have to go out of our way for this now. It feels like a more complex usecase and I would say, lets wait for feedback from users, before exploring this.

Reply:
I'll let user create the issue if they need it instead of already creating one.